src/createServer.js

```diff
@@ -1,11 +1,164 @@
+
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  let expenses = [];
+  let users = [];
+
+  app.use(express.json());
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+    }
+
+    const newUser = {
+      id: Math.floor(Math.random() * 1000),
+      name,
+    };
+
+    users.push(newUser);
+    res.statusCode = 201;
+    res.send(newUser);
+  });
+
+  app.get('/users', (req, res) => {
+    if (!users.length) {
+      res.statusCode = 200;
+      res.send([]);
+    }
+    res.send(users);
+  });
+
+  const findById = (items, value) => {
+    return items.find((item) => item.id === Number(value));
+  };
+
+  app.get('/users/:userId', (req, res) => {
+    const foundUser = findById(users, req.params.userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+    }
+    res.sendStatus = 200;
+    res.send(foundUser);
+  });
+
+  app.patch('/users/:userId', (req, res) => {
+    const foundUser = findById(users, req.params.userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+    }
+
+    const { name } = req.body;
+
+    Object.assign(foundUser, { name });
+    res.send(foundUser);
+  });
+
+  const filterById = (items, id) => {
+    return items.filter((item) => item.id !== Number(id));
+  };
+
+  app.delete('/users/:userId', (req, res) => {
+    const filteredUsers = filterById(users, req.params.userId);
+
+    if (users.length === filteredUsers.length) {
+      res.sendStatus(404);
+    }
+    users = filteredUsers;
+    res.sendStatus(204);
+  });
+
+  app.post('/expenses', (req, res) => {
+    const { userId, spentAt, title, amount, category, note } = req.body;
```

Comment:
You need here only `userId` 
That's why better rewrite 👇 
```suggestion
    const { userId } = req.body;
```
and in `newExpense` you could just pass `...req.body`.